File: src/main/java/com/clinichub/entity/Patient.java
package com.clinichub.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "patient")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    private Integer age;

    @Column(length = 10)
    private String gender;

    @Column(length = 30)
    private String phone;

    @Column(nullable = false)
    private Boolean enabled = true;

    @Column(name = "created_at", nullable = false)
    private java.time.LocalDateTime createdAt;
}

File: flutter_app/lib/models/patient.dart
class Patient {
  final int id;
  final String name;
  final int? age;
  final String? gender;
  final String? phone;
  final bool enabled;
  final String createdAt;

  Patient({
    required this.id,
    required this.name,
    this.age,
    this.gender,
    this.phone,
    required this.enabled,
    required this.createdAt,
  });

  factory Patient.fromJson(Map<String, dynamic> json) => Patient(
    id: json['id'],
    name: json['name'],
    age: json['age'],
    gender: json['gender'],
    phone: json['phone'],
    enabled: json['enabled'],
    createdAt: json['createdAt'],
  );

  Map<String, dynamic> toJson() => {
    'name': name,
    'age': age,
    'gender': gender,
    'phone': phone,
  };
}
